### PR TITLE
feat(dockerfile): Recognize Dockerfile with prefix by default

### PR DESCRIPTION
**Pull request by itchyny**
In addition to Dockerfile.something, I'd like to add something.Dockerfile support to the default config. I think there's no standard for this and some people prefer Dockerfile as an extension.

Refs:

- [docker - How to name Dockerfiles - Stack Overflow](https://stackoverflow.com/a/26084010/1363488)
- [filetype.vim#L391 · vim/vim · GitHub](https://github.com/vim/vim/blob/ff78155aa1755aced96a3b343e81939c94aac721/runtime/filetype.vim#L391)
- Updates #3940

### CODE REVIEW

**Review by viceice — APPROVED** [review]

**Comment by itchyny** [comment]
My concern is this section https://github.com/renovatebot/renovate/blob/780ac8cfa2cd24f919f0477dcd0fe5dfd66e596e/docs/usage/configuration-options.md#filematch.

File: lib/manager/dockerfile/index.ts [hunk 1]
```diff
@@ -7,5 +7,5 @@ const language = LANGUAGE_DOCKER;
 export { extractPackageFile, language, updateDependency };
 
 export const defaultConfig = {
-  fileMatch: ['(^|/)Dockerfile$', '(^|/)Dockerfile\\.[^/]*$'],
+  fileMatch: ['(^|/)Dockerfile$', '(^|/)Dockerfile\\.[^/]*$', '(^|/)[^/]*\\.Dockerfile$'],
```
Comment (rarkins) [inline lib/manager/dockerfile/index.ts]:
```suggestion
  fileMatch: ['(^|/|\\.)Dockerfile$', '(^|/)Dockerfile\\.[^/]*$'],
```

Would this work instead?

**Review by viceice — CHANGES_REQUESTED** [review]
Docs should be adapted accordingly. Also needs a `yarn prettier-fix` run.

**Review by viceice — APPROVED** [review]

**Comment by itchyny** [comment]
Thank you.

**Pull request merged**